java-generator/core/src/main/java/io/fabric8/java/generator/Config.java

```diff
@@ -116,6 +118,41 @@ public Config(
     }
   }
 
+  public Config(
+      Boolean uppercaseEnums,
+      Boolean objectExtraAnnotations,
+      Boolean generatedAnnotations,
+      Boolean alwaysPreserveUnknown,
+      Map<String, String> packageOverrides,
+      List<String> filesSuffixes,
+      String serDatetimeFormat,
+      String deserDatetimeFormat) {
+    if (uppercaseEnums != null) {
```

Comment:
Sorry but this looks a bit odd to me, Why are we creating a new constructor every time we're introducing a new argument?

Reply:
> Why are we creating a new constructor every time we're introducing a new argument?

Because dropping the old constructors would be a breaking change as noted [here](https://github.com/fabric8io/kubernetes-client/issues/4829#issuecomment-1419025789), probably we should clean up from time to time and remove the unused constructors ...

>  is it possible to employ builder here?

I think we have it, but it doesn't solve the issue with the constructors AFAIU.
I think there is a margin for improvement though, if you know what to do I'll happily review a PR proposal 🙂